typescript/core/src/parse.ts

```diff
@@ -417,3 +417,27 @@ function parseDataEnd(reader: Reader, recordLength: number): TypedMcapRecord {
     dataSectionCrc,
   };
 }
+
+export function monoParseMessage(reader: Reader): TypedMcapRecord | undefined | null {
```

Comment:
I also prefer `maybeParseMessage` as a name, and a comment explaining why it exists.

Reply:
`maybeParseMessage` isn't the best name here IMO, the fact that it might return undefined isn't the important part, the key defining quality of this function is to be optimizable / monomorphic.